plugin/storage/kafka/marshaller.go

```diff
@@ -0,0 +1,45 @@
+// Copyright (c) 2018 The Jaeger Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package kafka
+
+import (
+	"github.com/apache/thrift/lib/go/thrift"
+
+	"github.com/jaegertracing/jaeger/model"
+	"github.com/jaegertracing/jaeger/model/converter/thrift/jaeger"
+)
+
+// Marshaller encodes a span into a byte array to be sent to Kafka
+type Marshaller interface {
+	Marshal(*model.Span) ([]byte, error)
+}
+
+type thriftMarshaller struct {
+	tProtocolFactory *thrift.TBinaryProtocolFactory
+}
+
+func newThriftMarshaller() *thriftMarshaller {
+	return &thriftMarshaller{thrift.NewTBinaryProtocolFactoryDefault()}
+}
+
+// Marshall encodes a span as a thrift byte array
+func (h *thriftMarshaller) Marshal(span *model.Span) ([]byte, error) {
+	thriftSpan := jaeger.FromDomainSpan(span)
```

Comment:
@yurishkuro @vprithvi @davit-y brought up a very good point that the thrift model doesn't have the process in the definition (https://github.com/jaegertracing/jaeger-idl/blob/master/thrift/jaeger.thrift#L51) (our internal kafka model does). We have a couple of options,
1) Add process to the thrift model
2) Pass a batch with 1 span instead
3) Write a json marshaller for now

I've told Davit to go ahead with option 3 for now until we come to a consensus on how to move forward.

Reply:
my preference is to land my proto PR and use that. You can marshal domain model directly into `[]byte`.

Reply:
proto model was merged

Reply:
modeel/ids_test.go shows example of to/from `[]byte` marshaling